x-pack/plugins/security_solution/public/resolver/view/panels/node_events.tsx

```diff
@@ -8,11 +8,33 @@ import React, { memo, useMemo } from 'react';
 import { i18n } from '@kbn/i18n';
 import { EuiBasicTableColumn, EuiButtonEmpty, EuiSpacer, EuiInMemoryTable } from '@elastic/eui';
 import { FormattedMessage } from 'react-intl';
+import { useSelector } from 'react-redux';
 import { StyledBreadcrumbs } from './panel_content_utilities';
-
 import * as event from '../../../../common/endpoint/models/event';
 import { ResolverEvent, ResolverNodeStats } from '../../../../common/endpoint/types';
-import { useReplaceBreadcrumbParameters } from '../use_replace_breadcrumb_parameters';
+import * as selectors from '../../store/selectors';
+import { ResolverState } from '../../types';
+import { StyledPanel } from '../styles';
+import { useNavigateOrReplace } from '../use_navigate_or_replace';
+import { PanelLoading } from './panel_loading';
+
+export function NodeEvents({ nodeID }: { nodeID: string }) {
+  const processEvent = useSelector((state: ResolverState) =>
+    selectors.processEventForID(state)(nodeID)
+  );
+  const relatedEventsStats = useSelector((state: ResolverState) =>
+    selectors.relatedEventsStats(state)(nodeID)
```

Comment:
This is `O(n)` or worse where `n` is the number of processes in the graph. We should refactor the data selector called `relatedEventsStats`

Reply:
NVM i was wrong. this is OK